cpp/src/arrow/render_api.cpp

```diff
@@ -6,20 +6,7 @@ namespace zilliz {
 namespace render {
 
 std::shared_ptr<arrow::Array>
-get_pointmap(std::shared_ptr<arrow::Array> arr_x, std::shared_ptr<arrow::Array> arr_y) {
-    auto x_length = arr_x->length();
-    auto y_length = arr_y->length();
-    auto x_type = arr_x->type_id();
-    auto y_type = arr_y->type_id();
-    assert(x_length == y_length);
-    assert(x_type == arrow::Type::UINT32);
-    assert(y_type == arrow::Type::UINT32);
-
-    auto input_x = (uint32_t *) arr_x->data()->GetValues<uint8_t>(1);
-    auto input_y = (uint32_t *) arr_y->data()->GetValues<uint8_t>(1);
-
-    auto output = pointmap(input_x, input_y, x_length);
-
+out_pic(std::pair<uint8_t* ,int64_t> output) {
```

Comment:
why pair?

Reply:
output png's size is required for setting arrow::array's bitmap